setup-scripts/health-checker.py

```diff
@@ -114,55 +114,86 @@ async def check_health():
         if bad:
-            # We append the entire record, so we can get the full context.
-            failed_records.append(rec)
-        passed_checks += 1
-
-    checks = passed_checks + failed_checks
-    if len(failed_records) > 0:
-        message = "Found {}/{} failed checks ({} critical) over the last {} hours!".format(failed_checks, checks,
-                                                                                           failed_critical, CHECK_HOURS)
-        file = discord.File(io.BytesIO(json.dumps(failed_records, indent=2).encode()), filename="failed_checks.log")
-        notifyTeam = failed_critical > 0
-        await send_msg(client, message, file=file, force_notify=notifyTeam)
-        return
+            failed_records.append(critical["checks"])
+
+    for verbose in json_verbose:
+        time_unaware = datetime.strptime(verbose['date'], '%Y-%m-%dT%H:%M:%S.%fZ')  # time in UTC
+        time = pytz.utc.localize(time_unaware)  # time with time zone
+        if time < time_limit:
+            continue
+        bad = False
+        for check in verbose['checks']:
+            verbose_checks_total += 1
+            if check['up'] == False:
+                verbose_checks_failed += 1
+                bad = True
+        if bad:
+            failed_records.append(verbose["checks"])
+
+    ################################################################################
+    ################ create a message
+    ################################################################################
+
+    message = ""
+    force_notify = False
+
+    if json_check["disabled"]:
+        message += "__Portal manually disabled!__ "
+        force_notify = True
+    elif res_check.status_code is not requests.codes.ok:
+        message += "__Portal down!!!__ "
+        force_notify = True
+    
+    if critical_checks_failed:
+        message += "{}/{} CRITICAL checks failed over the last {} hours! ".format(critical_checks_failed, critical_checks_total, CHECK_HOURS)
+    else:
+        message += "All {} critical checks passed. ".format(critical_checks_total)
+
+    if verbose_checks_failed:
+        message += "{}/{} verbose checks failed over the last {} hours! ".format(verbose_checks_failed, verbose_checks_total, CHECK_HOURS)
+    else:
+        message += "All {} verbose checks passed. ".format(verbose_checks_total)
+
+    if len(failed_records):
+        failed_records_file = json.dumps(failed_records, indent=2)
 
-    # Send an informational heartbeat if all checks passed but only if it's in
-    # the first CHECK_HOURS hours of the day, essentially the first call.
-    if datetime.now().hour < CHECK_HOURS:
-        await send_msg(client, "Health checks passed: {}/{}\n".format(passed_checks, checks))
+    # send a message if portal is down, there is a failures dump or just once daily (heartbeat) based on CHECK_HOURS
+    if force_notify or failed_records_file or datetime.now().hour < CHECK_HOURS:
```

Comment:
If you want to `force_notify` on portal down, that's fine. But if you want to later deduce that the portal is down based on the `force_notify` value, I would prefer to have a different name for that var.

Reply:
that's my bad on description

replaced "end a message if portal is down" with "send a message if we force notification"